File: c/translate2d.c
#include "geotranslation.h"

int main(int argc, char** argv) {
    double lat, lon, distance, bearing, lat2, lon2;

    if (argc != 5) {
        printf("Usage: translate2d <latitude> <longitude> <distance> <bearing>\n");
        return -1;
    }

    lat = atof(argv[1]);
    lon = atof(argv[2]);
    distance = atof(argv[3]);
    bearing = atof(argv[4]);

    translate2d(lat, lon, distance, bearing, &lat2, &lon2);

    printf("Starting point: %f,%f\n", lat, lon);
    printf("Ending point: %f,%f\n", lat2, lon2);

    return 0;
}

File: java/Translate2d.java
public class Translate2d {
    public static void main(String[] args) {
        if (args.length != 4) {
            System.out.println("Usage: translate2d <latitude> <longitude> <distance> <bearing>");
            return;
        }

        double lat = Double.parseDouble(args[0]);
        double lon = Double.parseDouble(args[1]);
        double distance = Double.parseDouble(args[2]);
        double bearing = Double.parseDouble(args[3]);

        LatLong ll = Geotranslation.translate2d(lat, lon, distance, bearing);

        System.out.println(String.format("Starting point: %f,%f", lat, lon));
        System.out.println(String.format("Ending point: %f,%f", ll.getLatitude(), ll.getLongitude()));
    }
}
